Jetsnack/app/src/main/java/com/example/jetsnack/ui/JetsnackNavGraph.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.jetsnack.ui
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.lifecycle.Lifecycle
+import androidx.navigation.NavBackStackEntry
+import androidx.navigation.NavHostController
+import androidx.navigation.NavType
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.navArgument
+import androidx.navigation.compose.navigate
+import androidx.navigation.compose.navigation
+import androidx.navigation.compose.rememberNavController
+import com.example.jetsnack.ui.MainDestinations.SNACK_ID_KEY
+import com.example.jetsnack.ui.home.HomeSections
+import com.example.jetsnack.ui.home.home
+import com.example.jetsnack.ui.snackdetail.SnackDetail
+
+/**
+ * Destinations used in the ([JetsnackApp]).
+ */
+object MainDestinations {
+    const val HOME_ROUTE = "home"
+    const val SNACK_DETAIL_ROUTE = "snack"
+    const val SNACK_ID_KEY = "snackId"
+}
+
+@Composable
+fun JetsnackNavGraph(
+    modifier: Modifier = Modifier,
+    navController: NavHostController = rememberNavController(),
+    startDestination: String = MainDestinations.HOME_ROUTE,
+) {
+    NavHost(
+        navController = navController,
+        startDestination = startDestination
+    ) {
+        navigation(
+            route = MainDestinations.HOME_ROUTE,
+            startDestination = HomeSections.FEED.route
+        ) {
+            home(
+                onSnackSelected = { snackId: Long, from: NavBackStackEntry ->
```

Comment:
I'm not sure on passing the `NavBackStackEntry` back up. The lifecycle thing feels like a bit of a workaround to me, is this recommended by the nav team?

Reply:
I copied it from the Owl sample. @JoseAlcerreca can you elaborate?